codegen/ace/src/comparison.rs

```diff
@@ -0,0 +1,64 @@
+use air_ir::{Operation, TraceAccess, Value};
+use std::cmp::Ordering;
+
+/// order by segment, offset, then column
+fn compare_access(a: &TraceAccess, b: &TraceAccess) -> Ordering {
+    let segment = a.segment.cmp(&b.segment);
+    if segment == Ordering::Equal {
+        let row = a.row_offset.cmp(&b.row_offset);
+        if row == Ordering::Equal {
+            a.column.cmp(&b.column)
+        } else {
+            row
+        }
+    } else {
+        segment
+    }
```

Comment:
Nit: Tuples implement `Ord` if all its components implement `Ord`. If you need a different ordering of the structs, this should be simpler
```suggestion
    (a.segment, a.column, a.row_offset).cmp(&(b.segment, b.column, b.row_offset))
```